src/main/java/com/faforever/client/fx/PlatformServiceImpl.java

```diff
@@ -67,11 +127,28 @@ public void stopFlashingWindow(String windowTitle) {
     User32.INSTANCE.FlashWindowEx(flashwinfo);
   }
 
+  /**
+   * @return The title of the foreground window, may be null!
+   */
   @Override
   public String getForegroundWindowTitle() {
-    HWND hwnd = User32.INSTANCE.GetForegroundWindow();
-    char[] textBuffer = new char[255];
-    User32.INSTANCE.GetWindowText(hwnd, textBuffer, 255);
-    return new String(textBuffer).trim();
+    if (!isOnWindows) {
+      return null;
+    }
+
+    HWND window = User32.INSTANCE.GetForegroundWindow();
+
+    if (window != null) {
+      char[] textBuffer = new char[255];
+      User32.INSTANCE.GetWindowText(window, textBuffer, 255);
+      return new String(textBuffer).trim();
+    } else {
+      return null;
+    }
```

Comment:
```
  if (window == null) {
     return null;
  }

  char[] textBuffer = new char[255];
  User32.INSTANCE.GetWindowText(window, textBuffer, 255);
  return new String(textBuffer).trim();
```

Reduces nesting and favors "guards" where the method returns on invalid conditions, leaving the "actual" code always at the end.